binstar_client/utils/projects/filters.py

```diff
@@ -0,0 +1,110 @@
+import fnmatch
+import logging
+from os.path import exists, isfile, join
+
+log = logging.getLogger('binstar.projects.upload')
+
+
+class NoIgnoreFileException(IOError):
+    def __init__(self, msg, *args, **kwargs):
+        self.msg = msg
+        super(NoIgnoreFileException, self).__init__(msg, *args, **kwargs)
+
+
+class FilterBase(object):
+    def __init__(self, *args, **kwargs):
+        raise NotImplemented
+
+    def can_filter(cls):
+        return True
+
+    def run(self, pfile):
+        raise NotImplemented
+
+    def update_metadata(self, metadata={}):
+        raise NotImplemented
+
+
+class VCSFilter(FilterBase):
+    '''
+    Version Control System Filtering
+    '''
+    def __init__(self, pfiles, *args, **kwargs):
+        self.pfiles = pfiles
+
+    def run(self, pfile):
+        if pfile.relativepath.startswith('.git/'):
```

Comment:
@malev I think that we could use a general `.anacondaignore` because this would be useful for the build upload as well.